### ISSUE TITLE
Spreading a tuple into an Annotation now produces an error

### ISSUE THREAD

**Issue by jasonk** [microsoft/pyright#6963]
**Describe the bug**

I've been using some code that unpacks a tuple into an `Annotation` to allow adding a pydantic `FieldInfo` along with a custom annotation to a type from the same set of arguments.  It's been working fine, but recently Pyright started to report an error for it that says "Unpack is not allowed in this context". I narrowed it down to  https://github.com/microsoft/pyright/commit/ba06108bcba03b41da745c73b26e22e98c474d00 which says "Added missing error for an illegal unpacked argument in type argument", but if it's actually an illegal argument I haven't been able to figure out *why* it's illegal, since it works just fine and mypy doesn't complain about it.

**Code or Screenshots**
```
from typing import Annotated

class Info:
    pass

def field():
    return [ Info(), Info() ]


class Thing:
    item_id: Annotated[str, *field()]
```

**VS Code extension or command-line**
I can repro this with the code above using the current HEAD from git:

```
~/repos/pyright main*
❯ git describe
1.1.345-14-g89cef4ecf

~/repos/pyright main*
❯ ./packages/pyright/index.js repro.py
/Users/jason/repos/pyright/repro.py
  /Users/jason/repos/pyright/repro.py:11:28 - error: Unpack is not allowed in this context
1 error, 0 warnings, 0 informations
```

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression. The `Annotated` type form requires a bunch of special-case logic because it acts unlike most type forms. In particular, it requires the the first subscript value is a type expression (which is limited to only expression forms that are allowed within a type annotation), but it allows arbitrary value expressions for all of the remaining subscript values. An unpack operator would normally be illegal in most type annotations, but it is allowed in this context.

This will be fixed in the next release. I've also updated our unit tests to cover this case, which should prevent similar regressions in the future.

**Comment by erictraut** [comment]
This is included in pyright 1.1.346, which I just published. It will also be included in a future release of pylance.
